### PR TITLE
Show parent pubspecs when a pubspec could not be found.

### PR DESCRIPTION
**Pull request by sigurdm**
Related to https://github.com/dart-lang/pub/issues/4127

### CODE REVIEW

File: lib/src/package.dart [hunk 1]
```diff
@@ -140,6 +140,9 @@ class Package {
   /// This mechanism can be used to avoid loading pubspecs twice. It can also be
   /// used to override a pubspec in memory for trying out an alternative
   /// resolution.
+  ///
+  /// [referringPubspec] is the path to the pubspec that includes this one in
+  /// the workspace.
   factory Package.load(
     String dir,
```
Comment (jonasfj) [inline lib/src/package.dart:147]: nit: documentation string refers to `packageDir`, but it's called `dir`. Probably the same, but would be neat to fix. (unrelated ofcourse no need to do that in this PR)

File: lib/src/package.dart [hunk 2]
```diff
@@ -140,6 +140,9 @@ class Package {
   /// This mechanism can be used to avoid loading pubspecs twice. It can also be
   /// used to override a pubspec in memory for trying out an alternative
   /// resolution.
+  ///
+  /// [referringPubspec] is the path to the pubspec that includes this one in
```
Comment (jonasfj) [inline lib/src/package.dart:144]: I don't see this argument passed?

**Review by jonasfj — COMMENTED** [review]
Looks fine, but you seem to have included a few extra lines not related to this PR. I assume this is about improving the error message?

File: lib/src/package.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/package.dart:144]: Ah yeah, this was from another strategy I tried first. Deleted

File: lib/src/package.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/package.dart:147]: Done

**Pull request merged**